src/game_renderer/GameRenderer.cpp

```diff
@@ -133,12 +135,17 @@ void GameRenderer::Render(std::shared_ptr<GameState> game_state) {
   glUniformMatrix4fv(current_program->getUniform("V"), 1, GL_FALSE,
                      glm::value_ptr(V.topMatrix()));
 
-  for (std::shared_ptr<Platform> platform : *level->getPlatforms()) {
-    if (!ViewFrustumCulling::IsCulled(platform->GetBoundingBox(), vfplane)) {
-      MV = platform->GetTransform();
-      glUniformMatrix4fv(current_program->getUniform("MV"), 1, GL_FALSE,
-                         glm::value_ptr(MV.topMatrix()));
-      platform->GetModel()->draw(current_program);
+  for (std::shared_ptr<GameObject> obj : *level->getObjects()) {
+    if (obj->isObstacle()) {
+      if (Platform* platform = dynamic_cast<Platform*>(obj.get())) {
```

Comment:
Since you already know the object type, and to avoid using raw pointers, I suggest using a std::static_pointer_cast to Obstacle instead of dynamic_cast.

You could also combine the note rendering in this loop so that you don't have to iterate level->getObjects() twice